### PR TITLE
Fix perf tests fixture firing even when skipped

### PR DESCRIPTION
**Pull request by underyx**

### CODE REVIEW

File: semgrep/tests/performance/test_semgrep_rules_repo_perf.py [hunk 1]
```diff
@@ -221,20 +221,27 @@
 
 
 @pytest.fixture(scope="session")
-def semgrep_rules_repo(tmp_path_factory):
+def semgrep_rules_repo(request, tmp_path_factory):
```
Comment (mschwager) [inline semgrep/tests/performance/test_semgrep_rules_repo_perf.py:224]: So, I'm kinda confused about the architecture here. From what I understand about this PR, we're performing these fixture operations when tests are *collected*, not when they're *run*. This gives us a performance hit because these fixtures take a while to run. Is that right?

Based on above, I have a few questions:

* Are these fixtures being run because they don't have the `benchmark` argument/fixture? That's why adding it below causes the fixtures to be avoided unless we're actually benchmarking?
* Why do we have two separate fixtures here that only appear to be used with each other? Could this be one fixture to simplify things?
* It seems we could leverage existing pytest functionality rather than checking this information ourselves. Whether that's something like adding `benchmark` here to only run this fixture during benchmarking, or leveraging `pytest.mark.parametrize` with `indirect=True` to run the fixture during test-setup instead of test-collection. Am I missing something that makes that not possible?

File: semgrep/tests/performance/test_semgrep_rules_repo_perf.py [hunk 2]
```diff
@@ -221,20 +221,27 @@
...
+        ["git", "checkout", "c3196b4"], cwd=repo_path  # May 16, 2020
     )
 
-    yield repo_path
+    return repo_path
 
 
 @pytest.fixture(params=RULE_PATHS.strip().splitlines())
-def semgrep_rules_rule(semgrep_rules_repo, request, tmp_path):
+def semgrep_rules_rule(semgrep_rules_repo, request, tmp_path, benchmark):
```
Comment (mschwager) [inline semgrep/tests/performance/test_semgrep_rules_repo_perf.py:244]: Does this avoid running the fixture unless we're benchmarking? The fact that this variable is unused makes me think there's a better way to solve this.

File: semgrep/tests/performance/test_semgrep_rules_repo_perf.py [hunk 3]
```diff
@@ -221,20 +221,27 @@
...
 
 @pytest.fixture(scope="session")
-def semgrep_rules_repo(tmp_path_factory):
+def semgrep_rules_repo(request, tmp_path_factory):
     repo_path = tmp_path_factory.mktemp("repo")
+
+    getoption = request.config.getoption
+    is_disabled = getoption("benchmark_skip") or getoption("benchmark_disable")
+    is_force_enabled = getoption("benchmark_enable") or getoption("benchmark_only")
+    if is_disabled and not is_force_enabled:
```
Comment (mschwager) [inline semgrep/tests/performance/test_semgrep_rules_repo_perf.py:230]: Doesn't pytest have some existing functionality to do this for us? My concern is that we're duplicating internal pytest logic here, and that we could be missing an edge case they cover or that things could change out from underneath us and our logic no longer matches theirs. Thoughts?

File: semgrep/tests/performance/test_semgrep_rules_repo_perf.py (hunk 1 above)
Comment (underyx) [reply semgrep/tests/performance/test_semgrep_rules_repo_perf.py:224]: Basically I think all the questions are answered by: I assumed skipped tests' fixtures would not be executed. Sorry about that :(

This PR is the fastest way I found to fix the CI failures that are repo-wide, cause I'm supposed to dedicate all my time to semgrep-app now. The alternative I see is reverting the benchmarks commit.

File: semgrep/tests/performance/test_semgrep_rules_repo_perf.py (hunk 2 above)
Comment (underyx) [reply semgrep/tests/performance/test_semgrep_rules_repo_perf.py:244]: Yep, this makes pytest-benchmark decide if it should run. I'd need to research alternative ways to accomplish the same.

File: semgrep/tests/performance/test_semgrep_rules_repo_perf.py (hunk 3 above)
Comment (underyx) [reply semgrep/tests/performance/test_semgrep_rules_repo_perf.py:230]: Yep, it is slightly scary. It's actually pytest-benchmark logic we're duplicating for the record.

**Review by mschwager — APPROVED** [review]
Accepting to expedite fixing the tests :+1:
